test/system/src/test/java/io/pravega/test/system/ControllerFailoverTest.java

```diff
@@ -211,6 +216,17 @@ public void sweepOverTest() throws InterruptedException, ExecutionException {
 
         StreamImpl stream1 = new StreamImpl(scope, stream);
 
+        // Initiate and Create RG instances
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, clientConfig);
+        for (int i = 0; i < loopLimit; i++) {
+            readerGroupManager.createReaderGroup(readerGroupName + i,
+                    ReaderGroupConfig.builder().stream(io.pravega.client.stream.Stream.of(scope, stream)).build());
+        }
+
+        for (int i = 0; i < loopLimit; i++) {
+            controller.startScale(stream1, segmentsToSeal, newRangesToCreate);
+        }
         Future<Boolean> status = controller.startScale(stream1, segmentsToSeal, newRangesToCreate);
```

Comment:
As discussed, one approach would be to ensure we invoke `io.pravega.controller.server.rpc.grpc.v1.ControllerServiceImpl#scale` multiple times, wait for the controller to ack that the request has been registered and a scale event has been written. Post this the controller service instance count can be reduced to 0 and back to 1. Post the failover the test can verify if the scale operation for the correct epoch(s) have been completed. (This will also make the test more consistent.)

Reply:
`ControllerServiceImpl` might not be visible in system tests as it's an internal to Controller.